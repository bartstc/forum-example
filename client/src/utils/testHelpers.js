import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsErr;
};