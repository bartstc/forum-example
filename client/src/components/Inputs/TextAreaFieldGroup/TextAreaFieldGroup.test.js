import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextAreaFieldGroup from './TextAreaFieldGroup';
import { checkProps } from '../../../utils/testHelpers';

describe('TextAreaFieldGroup', () => {

  let wrapper;
  const props = {
    name: 'email',
    placeholder: 'email',
    value: 'email@gmail.com',
    error: 'error',
    info: 'info',
    onChange: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<TextAreaFieldGroup {...props} />);
  });

  it('renders TextAreaFieldGroup without errors', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should NOT throw a warning', () => {
    const propsError = checkProps(TextAreaFieldGroup, props);
    expect(propsError).toBeUndefined();
  });

});

