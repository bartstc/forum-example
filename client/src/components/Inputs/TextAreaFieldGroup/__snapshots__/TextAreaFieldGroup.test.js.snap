// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaFieldGroup renders TextAreaFieldGroup without errors 1`] = `
<div
  className="form-group"
>
  <textarea
    className="form-control form-control-lg is-invalid"
    name="email"
    onChange={[MockFunction]}
    placeholder="email"
    value="email@gmail.com"
  />
  <small
    className="form-text text-muted"
  >
    info
  </small>
  <div
    className="invalid-feedback"
  >
    error
  </div>
</div>
`;

exports[`renders TextAreaFieldGroup without errors 1`] = `
<div
  className="form-group"
>
  <textarea
    className="form-control form-control-lg is-invalid"
    name="email"
    onChange={[MockFunction]}
    placeholder="email"
    value="email@gmail.com"
  />
  <small
    className="form-text text-muted"
  >
    info
  </small>
  <div
    className="invalid-feedback"
  >
    error
  </div>
</div>
`;
